### ISSUE TITLE
[1.4 RC2] 'nodetool rebuild' fails under high load conditions

### ISSUE THREAD

**Issue by lmr** [scylladb/scylla#1776]
_Installation details_
Scylla version (or git commit hash): 1.4 RC2 AMI
Cluster size:
OS (RHEL/CentOS/Ubuntu/AWS AMI): 6 nodes, i2.2xlarge

Running longevity with large data set on a 6 node i2.2xlarge cluster, we had one instance where we failed to execute 'nodetool rebuild' in one of the nodes:

```
2016-10-19 05:02:20,257 nemesis          L0108 ERROR| sdcm.nemesis.ChaosMonkey: nodetool command 'nodetool -h localhost rebuild' failed on node Node scylla-1-4-rc2-large-data-scylla-db-node-321c45ec-5 [54.193.41.107 | 172.31.26.9] (seed: False): 
Command: /bin/ssh -a -x  -ControlPath=/var/tmp/ssh-masterMP90t4/socket -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpG2uR0K -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l centos -p 22 -i /var/tmp/scylla-1-4-rc2-large-data-longevity-test-de3cd72c.pem 54.193.41.107 "nodetool -h localhost rebuild"
Exit status: 1
Duration: 1.74153399467
Stdout:

Stderr:
nodetool: Failed to connect to 'localhost:7199' - ConnectException:
'Connection refused'.
```

Unfortunately, this condition seems difficult to reproduce. I logged into the node that presented this problem on the live cluster and was able to run 'nodetool status' just fine. I'm leaving this bug here for future reference - Apparently under high load conditions, the JMX server might just refuse to answer to requests.

**Comment by asias** [comment]
"nodetool -h localhost rebuild" will contact the scylla-jmx as well. if the jmx server is not responding the cmd will definitely fail.

**Comment by asias** [comment]
Just to be clear, it fails to run the rebuild cmd. It is not the rebuild process itself fails.

**Comment by asias** [comment]
@amnonh

**Comment by slivne** [comment]
all of this issues seem to be related to the fact that the jmx process is not restarted if scylla is killed 

I reproduced this on the AMI using the sinstruction that lucas provided

1. started a cluster of 6 servers - 3 should be enough in my view (i2.2x)
2. started a loader (c4.8x)
3. ran c-s according to the command provided
4. I waited for the load to run a bit (5 minutes)
5. entered a server
6. found the pid of scylla and killed only it ```sudo kill -9 <id>```
7. the scylla-jmx does not come up (even after waiting 15 minutes)

```
Oct 30 15:50:34 ip-172-30-0-149 scylla[2392]:  [shard 5] compaction - Compacting [/var/lib/scylla/data/ks2/standard1-19ab33009eb811e68c7d000000000003/ks2-standard1-ka-37-Data.db:level=0, /var/lib/scylla/data/ks2
Oct 30 15:50:50 ip-172-30-0-149 scylla[2392]:  [shard 5] compaction - Compacted 4 sstables to [/var/lib/scylla/data/ks2/standard1-19ab33009eb811e68c7d000000000003/ks2-standard1-ka-69-Data.db:level=0, ]. 94138259
Oct 30 15:51:22 ip-172-30-0-149 scylla[2392]:  [shard 2] compaction - Compacting [/var/lib/scylla/data/ks2/standard1-19ab33009eb811e68c7d000000000003/ks2-standard1-ka-18-Data.db:level=0, /var/lib/scylla/data/ks2
Oct 30 15:51:34 ip-172-30-0-149 sudo[3737]:   centos : TTY=pts/0 ; PWD=/home/centos ; USER=root ; COMMAND=/bin/kill -9 2392
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-server.service: main process exited, code=killed, status=9/KILL
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Stopped Run Scylla Housekeeping daily.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Stopping Run Scylla Housekeeping daily.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Stopping Scylla JMX...
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Unit scylla-server.service entered failed state.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-server.service failed.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Requested transaction contradicts existing jobs: Resource deadlock avoided
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-server.service holdoff time over, scheduling restart.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Requested transaction contradicts existing jobs: Transaction is destructive.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-server.service failed to schedule restart job: Transaction is destructive.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Unit scylla-server.service entered failed state.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-server.service failed.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-jmx.service: main process exited, code=exited, status=143/n/a
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Stopped Scylla JMX.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Unit scylla-jmx.service entered failed state.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-jmx.service failed.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Starting Scylla Server...
Oct 30 15:51:34 ip-172-30-0-149 scylla_prepare[3748]: Setting a physical interface eth0...
Oct 30 15:51:34 ip-172-30-0-149 scylla_prepare[3748]: Setting mask 00000011 in /proc/irq/100/smp_affinity
Oct 30 15:51:34 ip-172-30-0-149 scylla_prepare[3748]: Setting mask 00000022 in /proc/irq/101/smp_affinity
Oct 30 15:51:34 ip-172-30-0-149 scylla_prepare[3748]: Setting mask 00000044 in /proc/irq/102/smp_affinity
Oct 30 15:51:34 ip-172-30-0-149 scylla_prepare[3748]: Setting mask 00000011 in /sys/class/net/eth0/queues/tx-0/xps_cpus
Oct 30 15:51:34 ip-172-30-0-149 scylla_prepare[3748]: Setting mask 00000022 in /sys/class/net/eth0/queues/tx-1/xps_cpus
Oct 30 15:51:35 ip-172-30-0-149 scylla[3783]: Scylla version 1.4.rc3-20161028.e87bed5 starting ...
Oct 30 15:51:35 ip-172-30-0-149 collectd[2651]: network plugin: Ignoring notification with unknown severity 0.
```

```
[centos@ip-172-30-0-149 ~]$ sudo systemctl status scylla-jmx.service
● scylla-jmx.service - Scylla JMX
   Loaded: loaded (/usr/lib/systemd/system/scylla-jmx.service; disabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Sun 2016-10-30 15:51:34 UTC; 14min ago
  Process: 2409 ExecStart=/usr/lib/scylla/jmx/scylla-jmx -l /usr/lib/scylla/jmx (code=exited, status=143)
 Main PID: 2409 (code=exited, status=143)

Oct 30 15:30:09 ip-172-30-0-149 scylla-jmx[2409]: Using config file: /etc/scylla/scylla.yaml
Oct 30 15:30:11 ip-172-30-0-149 scylla-jmx[2409]: Connecting to http://127.0.0.1:10000
Oct 30 15:30:11 ip-172-30-0-149 scylla-jmx[2409]: Starting the JMX server
Oct 30 15:30:11 ip-172-30-0-149 scylla-jmx[2409]: JMX is not enabled to receive remote connections.
Oct 30 15:30:12 ip-172-30-0-149 scylla-jmx[2409]: service:jmx:rmi://localhost/jndi/rmi://localhost:7199/jmxrmi
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Stopping Scylla JMX...
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-jmx.service: main process exited, code=exited, status=143/n/a
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Stopped Scylla JMX.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: Unit scylla-jmx.service entered failed state.
Oct 30 15:51:34 ip-172-30-0-149 systemd[1]: scylla-jmx.service failed.
```

checking the scylla-jmx.service definition

```
cat /usr/lib/systemd/system/scylla-jmx.service
[Unit]
Description=Scylla JMX
Requisite=scylla-server.service
After=scylla-server.service
BindsTo=scylla-server.service

[Service]
Type=simple
EnvironmentFile=/etc/sysconfig/scylla-jmx
User=scylla
Group=scylla
ExecStart=/usr/lib/scylla/jmx/scylla-jmx -l /usr/lib/scylla/jmx
KillMode=process
Restart=on-abnormal

[Install]
WantedBy=multi-user.target
```

**Comment by slivne** [comment]
@syuu1228 can you take a look at this as well

**Comment by asias** [comment]
@slivne good catch! We need some systemd magic to to restart scylla-jmx whenever scylla-server is restarted.

**Comment by amnonh** [comment]
On Mon, Oct 31, 2016 at 4:24 AM, Asias He <notifications@github.com> wrote:

> @slivne <https://github.com/slivne> good catch! We need some systemd
> magic to to restart scylla-jmx whenever scylla-server is restarted.
>
What I think is happening:
After you kill scylla (not stopping the service) the scylla jmx fails and
shut itself down.
The scylla jmx service just start the scylla, but as there is no start
event of the service it does not try to start the jmx.

@syuu1228 <https://github.com/syuu1228> should we change the service file
of the jmx?

**Comment by avikivity** [comment]
We already have in `scylla-jmx.service`: `BindsTo=scylla-server.service`.  So if scylla-server is down, scylla-jmx should go down as well.

**Comment by syuu1228** [comment]
I will check systemd unit files are correctly implemented (or not)

**Comment by avikivity** [comment]
As far as I understand it is correct, but I probably don't understand correctly.

**Comment by syuu1228** [comment]
Looks like systemd does not tries to restart scylla-jmx when 'kill -9 <pid of scylla-server>', at least on CentOS7 AMI (or it takes too long, it not tries to restart at least in few minutes) . And it can reproduce on single node config.

**Comment by avikivity** [comment]
What did you kill?  scylla-jmx or scylla?

**Comment by syuu1228** [comment]
@avikivity Only scylla. When I killed it, also scylla-jmx.service stopped as 'failed' status, then it's not restart again.

**Comment by avikivity** [comment]
@syuu1228 maybe changed WantedBy=scylla-server.service to RequiredBy=? just guessing.

**Comment by avikivity** [comment]
btw, we should add `RequiresMountsFor=/var/lib/scyla` to make sure the filesystem is mounted before starting the service.

**Comment by amnonh** [comment]
On Mon, Oct 31, 2016 at 11:13 AM, Avi Kivity <notifications@github.com>
wrote:

> @syuu1228 <https://github.com/syuu1228> maybe changed
> WantedBy=scylla-server.service to RequiredBy=? just guessing.
>
No, that means that if for some reason the jmx will not run, scylla will
not start

there should be an indication for the service to retry jmx on fail

**Comment by avikivity** [comment]
Ah, we should change `Restart=on-abnormal` to `Restart=on-failure`

**Comment by avikivity** [comment]
@amnonh scylla-jmx exits when it can't connect to scylla, yes?  with what exit code?

**Comment by amnonh** [comment]
On Mon, Oct 31, 2016 at 11:22 AM, Avi Kivity <notifications@github.com>
wrote:

> @amnonh <https://github.com/amnonh> scylla-jmx exits when it can't
> connect to scylla, yes? with what exit code?
>
I think, it's no longer true, it used to be so when the jmx did polling.
But now, when it runs only when it is been used, it should be shutdown by
the service.

it would report an error if used with no scylla connection, but should not
shutdown.

**Comment by avikivity** [comment]
private static final class CheckRegistration extends TimerTask {
        private int missed_response = 0;
        // After MAX_RETRY retry we assume the API is not available
        // and the jmx will shutdown
        private static final int MAX_RETRY = 30;
        @Override
        public void run() {
            try {
                if (checkRegistration()) {
                    missed_response = 0;
                } else {
                    if (missed_response++ > MAX_RETRY) {
                        System.err.println("API is not available, JMX is shuting down");
                        System.exit(-1);
                    }
                }
            } catch (Exception e) {
                // ignoring exceptions, will retry on the next interval
            }
        }
    }

**Comment by slivne** [comment]
the user may change the path that scylla is located on via the config and
then will not be able to start

**Comment by amnonh** [comment]
Should no longer be used

**Comment by avikivity** [comment]
Then why did it enter the failed state?

**Comment by avikivity** [comment]
@slivne no

**Comment by amnonh** [comment]
@syuu1228 <https://github.com/syuu1228> could that be that it's due to the
change to: Requisite

**Comment by syuu1228** [comment]
On Mon, Oct 31, 2016 at 6:21 PM, Avi Kivity <notifications@github.com>
wrote:

> Ah, we should change Restart=on-abnormal to Restart=on-failure
>

Just tried. It able to restart jmx sometimes, but fails to restart sometime
too.
Probably it only success when scylla-server is restarted. When
scylla-server restart not completed, it prevent to restart, because of '
Requisite='

**Comment by amnonh** [comment]
On Mon, Oct 31, 2016 at 11:41 AM, Takuya ASADA <notifications@github.com>
wrote:

> On Mon, Oct 31, 2016 at 6:21 PM, Avi Kivity <notifications@github.com>
> wrote:
>
> > Ah, we should change Restart=on-abnormal to Restart=on-failure
> >
>
> Just tried. It able to restart jmx sometimes, but fails to restart sometime
> too.
> Probably it only success when scylla-server is restarted. When
> scylla-server restart not completed, it prevent to restart, because of '
> Requisite='

is it possible that the jmx restart too quickly and manage to do it before
scylla restart?

if scylla is not there, the jmx will not start

**Comment by avikivity** [comment]
@syuu1228 but we have After=scylla-server.service, so it should wait.

Maybe we have a conflict in the dependencies that causes systemd to ignore After=; or maybe scylla-server reports itself started before the API is ready.

**Comment by syuu1228** [comment]
After reverting the commit (
https://github.com/scylladb/scylla-jmx/commit/4672cd360f00880c38e91a3c2bdce84bcd876cec),
looks like systemd become able to restart scylla-jmx.

[centos@ip-172-30-0-216 ~]$ ps ax|grep scylla
  2569 pts/1    S+     0:00 grep --color=auto scylla-jmx
  3569 ?        Ssl    0:10 /usr/bin/scylla --log-to-syslog 1
--log-to-stdout 0 --default-log-level info --collectd-address=
127.0.0.1:25826 --collectd=1 --collectd-poll-period 3000 --network-stack
posix --num-io-queues 8 --max-io-requests 32 --cpuset
1,2,3,4,5,6,7,9,10,11,12,13,14,15
  3598 ?        Ssl    0:00 /usr/lib/scylla/jmx/symlinks/scylla-jmx
-Dcassandra.jmx.local.port=7199 -Xmx256m -XX:+UseSerialGC
-Dcom.sun.management.jmxremote.authenticate=false
-Dcom.sun.management.jmxremote.ssl=false -jar
/usr/lib/scylla/jmx/scylla-jmx-1.0.jar
  3630 pts/0    S+     0:00 grep --color=auto scylla
[centos@ip-172-30-0-216 ~]$ sudo kill -9 3569
[centos@ip-172-30-0-216 ~]$ ps ax|grep scylla
  2569 pts/1    S+     0:00 grep --color=auto scylla-jmx
  3680 ?        Rsl    0:09 /usr/bin/scylla --log-to-syslog 1
--log-to-stdout 0 --default-log-level info --collectd-address=
127.0.0.1:25826 --collectd=1 --collectd-poll-period 3000 --network-stack
posix --num-io-queues 8 --max-io-requests 32 --cpuset
1,2,3,4,5,6,7,9,10,11,12,13,14,15
  3709 ?        Ssl    0:00 /usr/lib/scylla/jmx/symlinks/scylla-jmx
-Dcassandra.jmx.local.port=7199 -Xmx256m -XX:+UseSerialGC
-Dcom.sun.management.jmxremote.authenticate=false
-Dcom.sun.management.jmxremote.ssl=false -jar
/usr/lib/scylla/jmx/scylla-jmx-1.0.jar
  3742 pts/0    S+     0:00 grep --color=auto scylla

**Comment by amnonh** [comment]
On Mon, Oct 31, 2016 at 11:45 AM, Avi Kivity <notifications@github.com>
wrote:

> @syuu1228 <https://github.com/syuu1228> but we have
> After=scylla-server.service, so it should wait.
>
> Maybe we have a conflict in the dependencies that causes systemd to ignore
> After=; or maybe scylla-server reports itself started before the API is
> ready.
>
The API should not be an issue, the jmx will not even try to call scylla
(unless someone is using nodetool) and even if it does, it should just
throw an error

**Comment by avikivity** [comment]
@syuu1228 see http://stackoverflow.com/questions/4192364/always-app-java-end-with-exit-143-ubuntu, we should add signal 143 to the list of clean signals  (`SuccessExitStatus=0 143`)

**Comment by asias** [comment]
I am not sure why it is so complex here. In my dev box, I just start the jmx once, e.g.,

java -Dapiaddress=127.0.0.1 -Djava.net.preferIPv4Stack=true -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=7199 -Dcom.sun.management.jmxremote.rmi.port=7199 -Dcom.sun.management.jmxremote.local.only=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -jar target/scylla-jmx-1.0.jar

and start/stop scylla all the time. The jmx works fine.

**Comment by avikivity** [comment]
@asias search for 143 in the logs above.  systemd send SIGTERM to jmx, java exited with 143, systemd thinks it's a failure (its not) and doesn't restart (because of on-abnormal)

So we need both on-failure and to mark signal 143 as a clean shutdown,

**Comment by syuu1228** [comment]
I'm not quite sure about this, but is this really not related to
"Requisite=" ?
man page of systemd.unit(5) says 'if the units listed here are not started
already, they will not be started and the transaction will fail
immediately'.
When scylla-jmx tries to restart before scylla-server restarted, maybe "it
will fail immediately", no?

https://www.freedesktop.org/software/systemd/man/systemd.unit.html

**Comment by syuu1228** [comment]
Tried SuccessExitStatus=0 143, but didn't change the result

**Comment by syuu1228** [comment]
I found that the issue can be avoid by adding "RequiredBy=scylla-server.service" on "[Install]" section.
Is it correct way to fix the issue, or we shouldn't add such dependency for scylla-server.service?

**Comment by amnonh** [comment]
On Tue, Nov 1, 2016 at 12:58 PM, Takuya ASADA <notifications@github.com>
wrote:

> I found that the issue can be avoid by adding "RequiredBy=scylla-server.service"
> on "[Install]" section.
> Is it correct way to fix the issue, or we shouldn't add such dependency
> for scylla-server.service?
>
Doesn't it mean that if (for some reason) the jmx fails, scylla-server will
not run?

**Comment by syuu1228** [comment]
On Tue, Nov 1, 2016 at 8:11 PM, Amnon Heiman <notifications@github.com>
wrote:

> On Tue, Nov 1, 2016 at 12:58 PM, Takuya ASADA <notifications@github.com>
> wrote:
>
> > I found that the issue can be avoid by adding "RequiredBy=scylla-server.
> service"
> > on "[Install]" section.
> > Is it correct way to fix the issue, or we shouldn't add such dependency
> > for scylla-server.service?
> >
> Doesn't it mean that if (for some reason) the jmx fails, scylla-server will
> not run?


Yes..I realized soon after sending previous mail, and also it means
restarting jmx triggers scylla-server restart.

**Comment by tzach** [comment]
@lmr is this an issue with 1.4.1?

**Comment by lmr** [comment]
Yes, it's still an issue:

    2016-11-10 11:37:20,268 nemesis          L0108 ERROR| sdcm.nemesis.CorruptThenRepairMonkey: nodetool command 'nodetool -h localhost repair' failed on node Node scylla-1-4-1-ld-corrupt-then-repair-scylla-db-node-1b5c79c6-5 [54.165.230.116 | 172.30.0.66] (seed: False): Command: /bin/ssh -a -x  -o ControlPath=/var/tmp/ssh-masterjZjxc3/socket -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpiFbrUL -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l centos -p 22 -i /var/tmp/scylla-1-4-1-ld-corrupt-then-repair-longevity-test-b274a7cf.pem 54.165.230.116 "nodetool -h localhost repair"
    Exit status: 1
    Duration: 1.07466101646
    Stdout:

    Stderr:
    nodetool: Failed to connect to 'localhost:7199' - ConnectException: 'Connection refused (Connection refused)'.

The situation seems better, as the first 3 rebuild attempts actually work, but after that we have 2 attempts that fail (the test is still in progress).

**Comment by slivne** [comment]
@lmr can you please provide the info on how to connect on the internal list - the issue here may be different then what we had before 

@lmr the log has ```CorruptThenRepairMonkey```  - and the issue is about rebuild - so is this the same case or a different one ?

**Comment by lmr** [comment]
@slivne, actually this happens for both, rebuild and repair. An example from the rebuild nemesis:

    2016-11-10 01:32:51,130 nemesis          L0108 ERROR| sdcm.nemesis.CorruptThenRebuildMonkey: nodetool command 'nodetool -h localhost rebuild' failed on node Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-d169bd2a-6 [54.218.123.104 | 172.31.46.234] (seed: False): Command: /bin/ssh -a -x  -o ControlPath=/var/tmp/ssh-masterlBJSu1/socket -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpYJ4oMY -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l centos -p 22 -i /var/tmp/scylla-1-4-1-ld-corrupt-then-rebuild-longevity-test-433670a0.pem 54.218.123.104 "nodetool -h localhost rebuild"
    Exit status: 1
    Duration: 1.34219312668
    Stdout:

    Stderr:
    nodetool: Failed to connect to 'localhost:7199' - ConnectException: 'Connection refused (Connection refused)'.

**Comment by lmr** [comment]
@slivne If you believe we need to open a separate issue for each admin command failing, I can duplicate this issue, saying 'repair' instead of 'rebuild'.

**Comment by lmr** [comment]
OK, so after I patched sct to wait until the JMX port is available before issuing the commands and the problem goes away. We have to wait from 1 to 3 minutes for the JMX service to be back up, from the data produced by the test suite:

    2016-11-11 05:08:38,811 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-2 [54.201.207.36 | 172.31.47.177] (seed: False): Waiting for JMX service to be up (60.2267241478 s)
    2016-11-11 05:37:46,806 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-3 [54.200.140.27 | 172.31.47.178] (seed: False): Waiting for JMX service to be up (60.2611529827 s)
    2016-11-11 06:26:33,971 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-5 [54.201.137.157 | 172.31.47.180] (seed: False): Waiting for JMX service to be up (60.102560997 s)
    2016-11-11 06:27:34,074 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-5 [54.201.137.157 | 172.31.47.180] (seed: False): Waiting for JMX service to be up (120.205533028 s)
    2016-11-11 06:28:34,327 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-5 [54.201.137.157 | 172.31.47.180] (seed: False): Waiting for JMX service to be up (180.459071875 s)
    2016-11-11 07:36:19,180 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-2 [54.201.207.36 | 172.31.47.177] (seed: False): Waiting for JMX service to be up (60.2749431133 s)
    2016-11-11 08:46:03,922 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-4 [54.212.221.80 | 172.31.47.179] (seed: False): Waiting for JMX service to be up (60.2892282009 s)
    2016-11-11 09:59:14,543 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-4 [54.212.221.80 | 172.31.47.179] (seed: False): Waiting for JMX service to be up (60.7950730324 s)
    2016-11-11 11:15:53,613 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-4 [54.212.221.80 | 172.31.47.179] (seed: False): Waiting for JMX service to be up (60.0819380283 s)
    2016-11-11 11:16:53,735 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-4 [54.212.221.80 | 172.31.47.179] (seed: False): Waiting for JMX service to be up (120.203937054 s)
    2016-11-11 11:17:53,922 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-4 [54.212.221.80 | 172.31.47.179] (seed: False): Waiting for JMX service to be up (180.391627073 s)
    2016-11-11 12:30:51,669 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-1 [54.191.51.159 | 172.31.47.175] (seed: False): Waiting for JMX service to be up (60.1133720875 s)
    2016-11-11 12:31:51,963 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-1 [54.191.51.159 | 172.31.47.175] (seed: False): Waiting for JMX service to be up (120.407465935 s)
    2016-11-11 13:53:50,793 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-3 [54.200.140.27 | 172.31.47.178] (seed: False): Waiting for JMX service to be up (60.0937259197 s)
    2016-11-11 13:54:51,005 wait             L0044 DEBUG| Node scylla-1-4-1-ld-corrupt-then-rebuild-scylla-db-node-6e969c62-3 [54.200.140.27 | 172.31.47.178] (seed: False): Waiting for JMX service to be up (120.305737019 s)

**Comment by lmr** [comment]
Given the previous fixes to the systemd service, and fixes to sct, this is no longer an issue. I'm closing it, thanks!
